middleware/session/middleware.go

```diff
@@ -0,0 +1,304 @@
+//
+// Usage:
+//
+//	app.Use(session.New())
+func New(config ...Config) fiber.Handler {
+	if len(config) > 0 {
+		handler, _ := NewWithStore(config[0])
+		return handler
+	}
+	handler, _ := NewWithStore()
+	return handler
+}
+
+// NewWithStore creates session middleware with an optional custom store.
+//
+// Parameters:
+//   - config: Variadic parameter to override default config.
+//
+// Returns:
+//   - fiber.Handler: The Fiber handler for the session middleware.
+//   - *Store: The session store.
+//
+// Usage:
+//
+//	handler, store := session.NewWithStore()
+func NewWithStore(config ...Config) (fiber.Handler, *Store) {
+	cfg := configDefault(config...)
+
+	if cfg.Store == nil {
+		cfg.Store = NewStore(cfg)
+	}
+
+	handler := func(c fiber.Ctx) error {
+		if cfg.Next != nil && cfg.Next(c) {
+			return c.Next()
+		}
+
+		// Acquire session middleware
+		m := acquireMiddleware()
+		m.initialize(c, cfg)
+
+		stackErr := c.Next()
+
+		m.mu.RLock()
+		destroyed := m.destroyed
+		m.mu.RUnlock()
+
+		if !destroyed {
+			m.saveSession()
+		}
+
+		releaseMiddleware(m)
+		return stackErr
+	}
+
+	return handler, cfg.Store
+}
+
+// initialize sets up middleware for the request.
+func (m *Middleware) initialize(c fiber.Ctx, cfg Config) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	session, err := cfg.Store.getSession(c)
+	if err != nil {
+		panic(err) // handle or log this error appropriately in production
+	}
+
+	m.config = cfg
+	m.Session = session
+	m.ctx = &c
+
+	c.Locals(middlewareContextKey, m)
+}
+
+// saveSession handles session saving and error management after the response.
+func (m *Middleware) saveSession() {
+	if err := m.Session.saveSession(); err != nil {
+		if m.config.ErrorHandler != nil {
+			m.config.ErrorHandler(m.ctx, err)
+		} else {
+			DefaultErrorHandler(m.ctx, err)
+		}
+	}
+
+	releaseSession(m.Session)
+}
+
+// acquireMiddleware retrieves a middleware instance from the pool.
+func acquireMiddleware() *Middleware {
+	m, ok := middlewarePool.Get().(*Middleware)
+	if !ok {
+		panic(ErrTypeAssertionFailed.Error())
+	}
+	return m
+}
+
+// releaseMiddleware resets and returns middleware to the pool.
+//
+// Parameters:
+//   - m: The middleware object to release.
+//
+// Usage:
+//
+//	releaseMiddleware(m)
+func releaseMiddleware(m *Middleware) {
+	m.mu.Lock()
+	m.config = Config{}
+	m.Session = nil
+	m.ctx = nil
+	m.destroyed = false
+	m.mu.Unlock()
+	middlewarePool.Put(m)
+}
+
+// FromContext returns the Middleware from the Fiber context.
+//
+// Parameters:
+//   - c: The Fiber context.
+//
+// Returns:
+//   - *Middleware: The middleware object if found, otherwise nil.
+//
+// Usage:
+//
+//	m := session.FromContext(c)
+func FromContext(c fiber.Ctx) *Middleware {
+	m, ok := c.Locals(middlewareContextKey).(*Middleware)
+	if !ok {
+		// TODO: since this may be called we may not want to log this except in debug mode?
+		log.Warn("session: Session middleware not registered. See https://docs.gofiber.io/middleware/session")
```

Comment:
I think the log is useful. Anyway, it might be still worth to add a config field to enable/disable logging

Reply:
log removed, logic for this: middleware that supports both legacy (Store) session API and Middleware API will need to check this and the log will be misleading and spammy.